src/tqec/detectors/__init__.py

```diff
@@ -0,0 +1,26 @@
+from .operation import (
+    ShiftCoords,
+    RelativeMeasurementData,
+    RelativeMeasurementsRecord,
+    Detector,
+    Observable,
```

Comment:
Do we want to expose `ShiftCoords`, `Detector` and `Observable` here? Not exposing them in the top-level API would be a good sign to tell to the user "if you have to dig into the library to get something, you should know what you are doing, so read the documentation first" in my opinion.

Reply:
Also, I am not sure that the `RelativeMeasurementsRecord` should be exposed. Does it have another usage than avoiding code duplication between `Observable` and `Detector`?